### ISSUE TITLE
Unexpected error during ptr inbounds test: cannot read from foreign (extern) static

### ISSUE THREAD

**Issue by kvark** [rust-lang/rust#71887]
Building gfx-rs on macOS with current nightly [errors out](https://travis-ci.org/github/gfx-rs/gfx/jobs/682984569#L3675):
```
error: internal compiler error: src/librustc_mir/interpret/validity.rs:425: Unexpected error during ptr inbounds test: cannot read from foreign (extern) static DefId(0:40 ~ dispatch[5c52]::ffi[0]::[0]::_dispatch_queue_attr_concurrent[0])

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:907:9

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.45.0-nightly (65b448273 2020-05-03) running on x86_64-apple-darwin

note: compiler flags: -C debuginfo=2 --crate-type lib

note: some of the compiler flags provided by cargo are hidden
```

Repro steps:
```bash
git clone https://github.com/gfx-rs/gfx
cd gfx
rustup override set nightly
make all
```

TODO: add a backtrace

**Comment by jonas-schievink** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/71612, should be fixed in the next nightly
